File: 26_remove_duplicates_from_sorted_array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int count = 0;
        for(int i = 0; i < nums.size(); i++){

            int k = i + 1;
            while(k < nums.size() && nums[k] == nums[i])
                k++;

            nums[count] = nums[i];
            i = k - 1;
            count++;   
        }
        return count;
    }
};

File: rs/26_remove_duplicates_from_sorted_array.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut count = 1;

        for index in 1..nums.len(){
            if nums[index] > nums[index - 1]{
                nums[count] = nums[index];
                count +=1;
            }
        }

        count as i32
    }
}